frontend/wallet/src/render/views/settings/DelegationSettings.re

```diff
@@ -71,20 +88,54 @@ let make = (~publicKey) => {
   let goBack = () =>
     ReasonReact.Router.push("/settings/" ++ PublicKey.uriEncode(publicKey));
 
-  // Mutation variables and handlers
-  let variables =
-    ChangeDelegation.make(
-      ~from=Apollo.Encoders.publicKey(publicKey),
-      ~to_=Apollo.Encoders.publicKey(PublicKey.ofStringExn(state.delegate)),
-      ~fee=
-        Apollo.Encoders.currency(
-          switch (state.fee) {
-          | DefaultAmount => defaultFee
-          | Custom(amount) => amount
-          },
-        ),
-      (),
-    )##variables;
+  let renderError = () => {
+    switch (state.error) {
+    | Some(EmptyDelegateKey) =>
+      <>
+        <Alert kind=`Danger defaultMessage="Please enter a public key" />
+        <Spacer height=2. />
+      </>
+    | Some(InvalidTransactionFee) =>
+      <>
+        <Alert
+          kind=`Danger
+          defaultMessage={"The minimum transaction fee is " ++ minimumFee}
+        />
+        <Spacer height=2. />
+      </>
+    | Some(InvalidGraphQLResponse(error)) =>
+      <> <Alert kind=`Danger defaultMessage=error /> <Spacer height=2. /> </>
+    | None => React.null
+    };
+  };
+
+  let onDelegateClick = (mutate: ChangeDelegationMutation.apolloMutation) => {
+    switch (state.delegate, state.fee) {
+    | ("", _) =>
+      changeState(prev => {...prev, error: Some(EmptyDelegateKey)})
+    | (_, Custom(fee)) =>
+      switch (fee) {
+      | "" =>
+        changeState(prev => {...prev, error: Some(InvalidTransactionFee)})
+      | fee when float_of_string(fee) < float_of_string(minimumFee) =>
+        changeState(prev => {...prev, error: Some(InvalidTransactionFee)})
+      | _ =>
```

Comment:
I tried to refactor this code duplication but I had problems with switch branches. If there are any thoughts, I'd love thoughts!

Reply:
it looks good to me!

Reply:
You actually don't need both statmenets, you can just remove the arrow on line 118 and line 119 and the first case will fall through to the second.

Reply:
When I try to remove the arrow on line 118, I get a syntax error stating
"Error: Variable fee must occur on both sides of this | pattern"
I believe it is due to the _when_ usage. I can refactor this nested switch into an if statement but I figured that the switch would be better.